### ISSUE TITLE
Leveraging Large Language Models for Robot 3D Scene Understanding

### ISSUE THREAD

**Issue by Shoichi-Hasegawa0628** [Shoichi-Hasegawa0628/summary_paper#41]
https://arxiv.org/abs/2209.05629
ICRA2023に投稿された論文

**Comment by Shoichi-Hasegawa0628** [comment]
Feed-Forward アプローチから

**どんなもの？ (Summary)**
3Dシーングラフにおいて高レベルなラベルを推論するために，
3つのアプローチを調査することで，
大規模言語モデルはロボットのシーン理解にどう使えるかを検討

① zero-shot
② feed-forwrd classifir
③ contrastive classfiier

![leverage](https://user-images.githubusercontent.com/74911522/199666497-c7c3f4be-c451-4936-ba92-f8026b3e28e7.JPG)

***************************************************************************

**どんな問題に取り組んだのか？**
3Dシーングラフのノードに「意味的なラベルを付けること」，
特に「部屋や建物のような高次の空間概念に対応するノード」は未解決の障害
⇒ システムがその部屋にどのようなオブジェクトがあるかを考慮しないといけない．
⇒⇒ そのような知識を提供する「常識的な」メカニズムが必要

**************************************************************************

**先行研究と比べてどこがすごい？**
・言語モデルの評価時や意味的な埋め込み時に、言語モデルに潜在する常識を活用するため、見知らぬ対象にも汎化でき、学習も不要もしくは最小限で済む
・含まれる部屋から建物のラベルを推定したり、実際の3Dシーングラフから部屋のラベルを推定するために使用することで、その適用性を示す
・3次元シーングラフに対する部屋ラベルの推定に用いることで、これらのアプローチの適用可能性


Metric-Semantic SLAM：
高いレベルの意味的なラベルは無視する
Hierarchical Mapping and Building Parsing：
3Dシーングラフの空間情報への意味付けは依然として，困難
Language and Robotics 
新しい部屋やオブジェクトに一般化しにくいまま

**************************************************************************

**技術や手法のキモはどこ？**
大規模言語モデルを活用した3つのアプローチを検討
**Zero-Shot**
事前に学習した言語モデルが、どの部屋のラベルがクエリに最も適しているかをスコアリング
- P(r_j | o_i)でエントロピーが低い物体をk個ピックアップ
(エントロピーが低いということは，その場所における物体を強く表しているから)
- そこからクエリ作って，場所を予測させるようにLLMに突っ込む
- LLMで各場所のスコアを出力
![zero-shot](https://user-images.githubusercontent.com/74911522/199666400-2d1763ef-93cb-419d-9c58-6a3feda072fb.JPG)

**Feed-Forward**
クエリの文字列をLLMで埋め込み，fine-tunedされたNNで部屋のラベルの分布を出力
- 「This room contains o1, o2, ...」のような文を作る
- LLMに突っ込んで埋め込みベクトルを得る (次元数は不明)
- そのベクトルをファインチューニングされたNNに突っ込み，場所ラベルごとのprediction logits (ソフトマックス関数に突っ込む前の重み付け和に相当するベクトル)を出力
- その最大値を指す場所ラベルをピックアップ
https://minus9d.hatenablog.com/entry/2020/10/25/193018
![feed-forward](https://user-images.githubusercontent.com/74911522/199666429-ec6d9b7a-7ff3-4941-ba4d-0b206200b170.JPG)

⇒ ほしい場所ラベルに焦点を向けた物体ラベルを含んだ文を作るから，フィードフォワードなのかな？

**Contrastive**
クエリと部屋ラベルの文字列は、微調整されたニューラルネットワークによって共
有された高次元空間に埋め込まれ、対応するクエリとラベルのペアが近くにマッピングされるように変換

- 「This room has o1, o2,...」と「This room is a bathroom」をLLMに突っ込み，埋め込み表現を得る
- 別々のファインチューニングNNにベクトルを入れて，場所ラベルごとのpredicate logitsを出す
- 二つのpredicate logitsでコサイン類似度かけて，高い類似の場所ラベルをピックアップ

![contrastive](https://user-images.githubusercontent.com/74911522/199666453-8e544dc9-1adf-4575-9ee6-87a4315d45c8.JPG)


共通して使うこと：
・部屋の中の全ての物体はクエリに入れず, k個の意味のある物体のみ使う．
⇒ doorやlightみたいなどこにでもありそうな物体は避ける
⇒⇒ そのときに，P(r_j | o_i)を求める．
　　　　・Gound-truth co-occurrencies：それぞれの物体ラベルがどの部屋で見られやすいか. ラプラススムージング (一様分布)
　　　　・言語モデルのクエリによる代理共起確率 (proxy co-occurrency probability)
　　　　p(rj |oi) = exp(logP(W_o_r)) / ∑ exp(log P(W_o_r))
↓
k個の物体の選び方はエントロピー計算して，曖昧さが少ない奴を選ぶためにarg minを取る

**************************************************************************

**何で有効性を確かめた？**
- Matterport3Dでシーングラフを作成
- 物体ラベルはmpcat40とnyuClassを使っている？

_比較手法_
・Baseline①：統計的なBaseline (GTの共起性を表現)
P(r_j | O) ≒ Π P(r_j | o_i)

・Baseline②：GraphSage (グラフニューラルネットワーク)
・Zero-shot
　・GT
　・Proxy (GPT-Jで物体選ぶ，k=3)

・Feed-Forward
　　・RoBERTa
　　　・GT
　　　・Proxy
　　・BERT-large
　　　・GT
　　　・Proxy

・Contrastive
　　・RoBERTa
　　　・GT
　　　・Proxy
　　・BERT-large
　　　・GT
　　　・Proxy

_タスク_
- 物体ラベルから部屋ラベルの推定

- 部屋ラベルから建物ラベルの推定 (Cotrastive approachはLbのスパース性が高いからやらない？)
　- GraphSage
　- Zero-shot (k=4)
　- Feed-Forward

- Real Scene Graph (Hydraで作成した動的なシーングラフ，3部屋)
　- zero-shot
　　- 物体ラベルは23ラベルを持つHRNetで推定


_結果：_
Accuracy (正解数 / データ数)
- Zero-Shot
　- lobbiesやfamily roomのような構成される物体が似ている (tables, chair)部屋は予測精度が悪い
　- Bar, librariesなど建物のようなは部屋はスコアが悪い (ジムはgym equipmentという明確な物体があったから精度は良い)

- Feed Forward & Contrastive
　- Zero-Shotよりは全体を通して良い
　- Proxyで物体をピックアップする方はZero-shotと対して変わらない

- Feed Forward vs Contrastive
　- どの物体が含まれているときに，部屋ラベルの予測精度が良いか


Building Trial Results
すべての手法で同じような結果

Real 3D Scene Graph Results
Zero-shotで行い，推定できた


*************************************************************************

**議論したいことは？**



*************************************************************************

**分からないところ**
mpcat40とnyuClassというのがよく分からない． ⇒ object label spaceとある
物体と場所のセット？
部屋と物体のセットの決め方がよく分からない


部屋のラベル付け：
全てのアルゴリズムがより大きなラベル空間から恩恵

言語の埋め込み：
・学習時に除外されたオブジェクトへの汎化
・学習時よりもはるかに大きなオブジェクト空間での推論が可能

建物のラベリング
提案アプローチはシーン理解へ一般的な適用性 ⇒ ベンチマーク必要

[3]で生成した実際のシーングラフ
Zero-shotラベリングアプローチは，スケーラブル・サンプル効率が良い．


*********************************************************************

**キーワードの定義**
・Contanairized roomってなんだ？

・大規模言語モデルは2種類に大きく分かれる
Masked Language Models (MLMs)：BERT，RoBERTa
Next-Token Prediction Models：GPT

・文字列Wから要約埋め込みの「言語モデルのマッピング」は，Λ(W)で表す．

・確率モデルにおける「スムージング」は，訓練データに出現しない事象に対し，微小な確率値を割り当てること．平滑化ともよばれる．
　・ラプラススムージング：事前分布に一様分布を仮定すること (αk≣1)

・ニューラルネットワークのスコアリングというのは，ソフトマックス関数で出力した確率のことなのかな



**興味のある論文**
